scheduler/src/main/java/com/flipkart/flux/eventscheduler/service/EventSchedulerService.java

```diff
@@ -0,0 +1,126 @@
+    private final InstrumentedScheduledExecutorService scheduledExecutorService;
+    private ObjectMapper objectMapper;
+
+    @Inject
+    public EventSchedulerService(EventSchedulerDao eventSchedulerDao, EventSchedulerRegistry eventSchedulerRegistry,
+                                 @Named("eventScheduler.batchRead.intervalms") Integer batchReadInterval,
+                                 @Named("eventScheduler.batchRead.batchSize") Integer batchSize,
+                                 ObjectMapper objectMapper) {
+        this.eventSchedulerDao = eventSchedulerDao;
+        this.eventSchedulerRegistry = eventSchedulerRegistry;
+        this.batchReadInterval = batchReadInterval;
+        this.batchSize = batchSize;
+        this.objectMapper = objectMapper;
+
+        ScheduledThreadPoolExecutor executor = new ScheduledThreadPoolExecutor(1);
+        // remove the task from scheduler on cancel
+        executor.setRemoveOnCancelPolicy(true);
+        scheduledExecutorService =
+                new InstrumentedScheduledExecutorService(Executors.unconfigurableScheduledExecutorService(executor),
+                        SharedMetricRegistries.getOrCreate(METRIC_REGISTRY_NAME), scheduledExectorSvcName);
+    }
+
+    public void start() {
+        synchronized (this) {
+            if (scheduledFuture == null || scheduledFuture.isDone()) {
+                scheduledFuture = scheduledExecutorService.scheduleWithFixedDelay(() -> {
+                    try {
+                        triggerEvents();
+                    }
+                    catch(Exception e) {
+                        logger.error("Error while running triggerEvents", e);
+                    }
+                }, initialDelay, batchReadInterval, TimeUnit.MILLISECONDS);
+                logger.info("EventSchedulerService started");
+            }
+        }
+    }
+
+    public void stop() {
+        synchronized (this) {
+            scheduledFuture.cancel(false);
+            logger.info("EventSchedulerService stopped");
+        }
+    }
+
+    public Boolean isRunning() {
+        return (scheduledFuture != null && !scheduledFuture.isDone());
+    }
+
+    private void triggerEvents() {
+        int offset = 0;
+        List<ScheduledEvent> events;
+        //time in seconds
+        long now = System.currentTimeMillis()/1000;
+        do {
+            events = eventSchedulerDao.retrieveOldest(offset, batchSize);
+            events.stream().filter(e -> e.getScheduledTime() <= now).forEach(e -> {
+                try {
+                    EventData eventData = objectMapper.readValue(e.getEventData(), EventData.class);
+                    eventSchedulerRegistry.triggerEvent(e.getEventName(), eventData.getData(), e.getCorrelationId(), eventData.getEventSource());
+                    eventSchedulerRegistry.deregisterEvent(e.getCorrelationId(), e.getEventName());
+                } catch (Exception ex) {}
+            });
+            offset += batchSize;
```

Comment:
shouldn't the offset be always 0, as we are deleting the rows from the database along. So the new query will not have old rows, and we don't need to increment the offset.